code/modules/reagents/chemistry/reagents/pyrotechnic.dm

```diff
@@ -56,10 +56,11 @@
 /datum/reagent/napalm/reaction_temperature(exposed_temperature, exposed_volume)
 	if(exposed_temperature > T0C + 100)
 		var/radius = min(max(0, volume * 0.15), 8)
-		fireflash_sm(get_turf(holder.my_atom), radius, rand(3000, 6000), 500)
+		var/turf/T = get_turf(holder.my_atom)
 		fire_flash_log(holder, id)
 		if(holder)
 			holder.del_reagent(id)
+		fireflash_sm(T, radius, rand(3000, 6000), 500)
```

Comment:
Why not just leave the `get_turf()` on the fireflash line? It's not being used anywhere else so I don't think it needs to be set as a variable.
```suggestion
 		fire_flash_log(holder, id)
 		if(holder)
 			holder.del_reagent(id)
		fireflash_sm(get_turf(holder.my_atom), radius, rand(3000, 6000), 500)

```
Same below as well.

Reply:
`holder` can be null at that point due to line 62. Removing the reagent from the holder will IIRC qdel the reagent.